prism/src/explicit/ProbModelChecker.java

```diff
@@ -1281,17 +1282,32 @@ public void exportStateRewardsToFile(Model model, int r, int exportType, PrismLo
 	 * @param exportType The format in which to export
 	 * @param out Where to export
 	 * @param precision number of significant digits >= 1
+	 * @param noexportheaders disables export headers for srew files
 	 */
-	public void exportStateRewardsToFile(Model model, int r, int exportType, PrismLog out, int precision) throws PrismException
+	public void exportStateRewardsToFile(Model model, int r, int exportType, PrismLog out, boolean noexportheaders, int precision) throws PrismException
 	{
 		int numStates = model.getNumStates();
 		int nonZeroRews = 0;
+		Rewards modelRewards;
 
 		if (exportType != Prism.EXPORT_PLAIN) {
 			throw new PrismNotSupportedException("Exporting state rewards in the requested format is currently not supported by the explicit engine");
 		}
 
-		Rewards modelRewards = constructRewards(model, r);
+		try {
+			modelRewards = constructRewards(model, r);
+		} catch (PrismException e) {
+			if (e.getMessage() == "Explicit engine does not yet handle transition rewards for D/CTMCs.") {
+				// add header to empty srew file
+				if (!noexportheaders) {
+					out.println("# State rewards");
+				}
+				out.println(numStates + " 0");
+				return;
+			} else {
```

Comment:
This seems wrong. If the rewards cannot be built due to the choice of engine, this is an error. Exporting a valid - but incorrect - rewards file creates confusion.

Reply:
Relying on the error message is fragile and hence should be avoided, but that's easy to fix. However, this code deals with the following problem: How should the export behave if transition rewards are present but not supported by the explicit engine?
- Before this PR, Prism aborted the export but wrote an _empty file_ resulting in some reward structures being exported properly and an empty file for the structure with transition rewards.
- With this PR, Prism also aborts but adds the default header to the offending file if not suppressed by `-noexportheaders`

If we do not continue this behaviour, I suggest to not write a file for the offending reward structure but skipping it (and warn) while still increasing the file counter. This exports all reward structures that have no transition rewards while preserving the same file names as the export with the symbolic engine. Import with `.all` would fail which prevents the wrong impression of everything being allright.

Reply:
[diff.patch](https://github.com/prismmodelchecker/prism/files/10880636/diff.patch)

Reply:
Yes, so that old behaviour was wrong. This is now fixed in master 67761e7a (thanks @merkste ) and I'll remove the code cited above as I review the PR.